internal/mail/rfc2822/headers.go

```diff
@@ -106,3 +115,16 @@ func parseContentType(h nm.Header) string {
 
 	return sources[0]
 }
+
+func parsePublicKey(h nm.Header) (crypto.PublicKey, error) {
+	sources, ok := h["Public-Key"]
+	if !ok {
+		return nil, errors.Errorf("header missing")
+	}
+
+	if len(sources) == 0 {
+		return nil, errors.Errorf("empty header")
+	}
+
+	return secp256k1.PublicKeyFromHex(sources[0])
```

Comment:
This implementation would only work with `secp256k1` there are other key types try looking at the `multikey` package

Reply:
@robdefeo using ```multikey.PublicKeyFromBytes()```, one can get the ```PublicKey```, but it requires a parameter ```keyType```. How can the ```keyType``` be inferred, when we only have the string representation of the public address?

Reply:
Hi @Kifen the `keyType` will be available as one of the headers e.g. `"Public-Key-Type: secp256k1\r\n"` this header can be parsed and the value passed to `multikey.PublicKeyFromBytes()`